tests/test_core.py

```diff
@@ -32,3 +33,10 @@ def t():
 
 def test_that_wait_until_with_zero_timeout_tries_once():
     assert stbt.wait_until(lambda: True, timeout_secs=0)
+
+
+def test_wait_until_consecutive_secs():
```

Comment:
Could probably do with some more tests here.  `wait_until` is becoming a little complex with multiple interacting parts.  It's good to centralise this complexity in one place but it's also good to sure how it works.

Perhaps some tests that mock out `time.time`?  Something like:

```
@contextmanager
def faketime_fn(sequence):
    it = iter(sequence)
    t = [0]
    def fake_time():
        return t[0]
    def fn():
        v = it.next()
        t[0] += 0.1
        return v
    with mock.mock("time.time", faketime):
        yield fn

def test():
    with faketime_fn([0,1,2,3,4,5,5,5,5,5,6,7,8,9,10]) as fn:
        assert wait_until(fn, consecutive_secs=0.5)
    with faketime_fn([0,1,2,3,4,5,5,5,5,5,6,7,8,9,10]) as fn:
        assert not wait_until(fn, consecutive_secs=0.6)
```